### ISSUE TITLE
Error Widget UI has changed worser than red screen

### ISSUE THREAD

**Issue by akhilpune** [jhomlala/catcher#91]
Until yesterday the error widget Ui was fine.
But now after update the error widget Ui has changed to this
![Screenshot_20200508-224353](https://user-images.githubusercontent.com/26247742/81431237-77339f00-917e-11ea-81ca-bad9e138bab6.jpg)

**Comment by jhomlala** [comment]
Hi, can you confirm that

**Comment by visual-velocity** [comment]
I'm also getting this issue now. How could we solve it

**Comment by jhomlala** [comment]
Again, please provide more details. I can't help you if you don't provide any helpful informations.
Please check error_widget_example.dart in example module.  Error widget seems to be working after flutter update. Maybe there's something wrong with your code. Please share some example.

**Comment by jhomlala** [comment]
Please open new ticket if this error still occurs. Please provide example code.

**Comment by SaravanarajaGITHUB** [comment]
Facing the same issue 

`class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      navigatorKey: Catcher.navigatorKey,
      builder: (BuildContext context, Widget widget) {
        Catcher.addDefaultErrorWidget(showStacktrace: false, title: "Custom error title", description: "Custom error description", maxWidthForSmallMode: 150);
        return widget;
      },
      title: 'Flutter Demo',
      theme: ThemeData(

        primarySwatch: Colors.blue,

        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}`

**Comment by jeiea** [comment]
Can't we make the `CatcherErrorWidget` work without a scaffold?
Otherwise we should separate the scaffold from complex code for protecting it from widget build failure, and that's somewhat irritating.

**Comment by Samurai016** [comment]
@jhomlala I think this issue should be reopen or at least updated, I'm facing the same issue.  
Is there a way to make `CatcherErrorWidget` working without a scaffold as @jeiea suggested?
For example, I have my scaffold surrounded by a `WillPopScope` widget and I can't use `CatcherErrorWidget`.

**Comment by Samurai016** [comment]
I found a solution that could work for everyone.
I noticed that the problem is that the rendering method [`_buildNormalErrorWidget()`](https://github.com/jhomlala/catcher/blob/master/lib/utils/catcher_error_widget.dart#L43) return a `Container` widget, which is displayed as the black screen with the red text as @akhilpune noticed.
I explain you my solution in some steps.

### `CatcherErrorWidget` class
I simply copied the [`CatcherErrorWidget`](https://github.com/jhomlala/catcher/blob/master/lib/utils/catcher_error_widget.dart) class into a custom version of it (I called it `CustomErrorWidget`).  

The class is the same as the original one (I just put the default values inside the constructor), excepting for the `_buildNormalErrorWidget` method, where I surrounded the `Container` with a `Scaffold`.
> You can also provide to the Scaffold an appbar and everything you want according to you app design.
``` dart
Widget _buildNormalErrorWidget() {
    return Scaffold(
      body: Container(
        margin: const EdgeInsets.all(20),
        child: Center(
          child: ListView(children: [
            _buildIcon(),
            Text(
              title,
              style: const TextStyle(color: Colors.black, fontSize: 25),
              textAlign: TextAlign.center,
            ),
            const SizedBox(height: 10),
            Text(
              _getDescription(),
              textAlign: TextAlign.center,
            ),
            const SizedBox(height: 10),
            _buildStackTraceWidget()
          ]),
        ),
      ),
    );
  }
```

### `builder` property in `MaterialApp`
To enable my custom widget I changed the `builder` property in my `MaterialApp` from [the default implementation provided by catcher](https://github.com/jhomlala/catcher/blob/master/example/lib/error_widget_example.dart#L34) to the my one:
``` dart
...
      builder: (BuildContext context, Widget? widget) {
        ErrorWidget.builder = (FlutterErrorDetails details) {
          return CustomErrorWidget(
            details: details,
            // Here you can pass your custom settings
          );
        };
        return widget!;
      },
...
```

Here you are, everything works.

# TL;DR
* Copy and paste [this class](https://gist.github.com/Samurai016/7186bdd214f2fef4a40ec5cf4f4098b2) in your code.
* Add this to your `MaterialApp` (instead of [this](https://github.com/jhomlala/catcher/blob/master/example/lib/error_widget_example.dart#L34))
   ``` dart
      builder: (BuildContext context, Widget? widget) {
        ErrorWidget.builder = (FlutterErrorDetails details) {
          return CustomErrorWidget(
            details: details,
            // Here you can pass your custom settings
          );
        };
        return widget!;
      },
   ```
